transport/handler.go

```diff
@@ -37,3 +37,17 @@ type Handler interface {
 		resw ResponseWriter,
 	) error
 }
+
+// OnewayHandler handles a single, transport-level, oneway request.
+type OnewayHandler interface {
+	// Handle the given oneway request
+	//
+	// An error may be returned in case of failures. BadRequestError must be
+	// returned for invalid requests. All other failures are treated as
+	// UnexpectedErrors.
+	HandleOneway(
+		ctx context.Context,
+		opts Options,
+		req *Request,
+	) error
```

Comment:
Is the plan to have these the actual execution of the oneway function in a goroutine?  If so, at which point will we be starting the goroutine (transport level, encoding level, right before the procedure)?

All to the point of, will we need the error return?